File: packages/ash-php/src/Core/BuildProofInput.php
<?php

declare(strict_types=1);

namespace Ash\Core;

/**
 * Input for building a proof.
 */
final class BuildProofInput
{
    /**
     * @param AshMode $mode ASH mode
     * @param string $binding Canonical binding: 'METHOD /path'
     * @param string $contextId Server-issued context ID
     * @param string $canonicalPayload Canonicalized payload string
     * @param string|null $nonce Optional server-issued nonce
     */
    public function __construct(
        public readonly AshMode $mode,
        public readonly string $binding,
        public readonly string $contextId,
        public readonly string $canonicalPayload,
        public readonly ?string $nonce = null,
    ) {
    }
}

File: packages/ash-dotnet/src/Ash.Core/BuildProofInput.cs
// ASH was developed by 3maem Co. | 12/31/2025

namespace Ash.Core;

/// <summary>
/// Input for building a proof.
/// </summary>
public sealed class BuildProofInput
{
    /// <summary>
    /// ASH mode.
    /// </summary>
    public required AshMode Mode { get; init; }

    /// <summary>
    /// Canonical binding: 'METHOD /path'.
    /// </summary>
    public required string Binding { get; init; }

    /// <summary>
    /// Server-issued context ID.
    /// </summary>
    public required string ContextId { get; init; }

    /// <summary>
    /// Canonicalized payload string.
    /// </summary>
    public required string CanonicalPayload { get; init; }

    /// <summary>
    /// Optional server-issued nonce.
    /// </summary>
    public string? Nonce { get; init; }
}
